spring-client-common/src/main/java/io/camunda/commons/auth/SelfManagedAuthentication.java

```diff
@@ -0,0 +1,115 @@
+package io.camunda.commons.auth;
+
+import org.apache.hc.client5.http.classic.methods.HttpPost;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.UnsupportedEncodingException;
+import java.lang.invoke.MethodHandles;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static io.camunda.commons.http.JsonUtils.toResult;
+
+public class SelfManagedAuthentication extends JwtAuthentication {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  private String authUrl;
+  private String keycloakRealm = "camunda-platform";
```

Comment:
For a self-managed setup, there should be other IDPs available soon. An abstraction could be handy here and help to keep the code maintainable

Reply:
Not still familiar with IDP and Identity world. I would meet with Identity team soon and will check more about upcoming IDPs.

Adding a TODO for now in https://github.com/camunda-community-hub/spring-zeebe/pull/510/commits/0922e0fe0feb5c4c0f8a6339d7d99b79ecfeefc1